src/test/ui/async-await/issue-98634.rs

```diff
@@ -45,6 +45,5 @@ fn main() {
         StructAsync { callback }.await;
         //~^ ERROR expected `fn() -> impl Future<Output = ()> {callback}` to be a fn item that returns `Pin<Box<(dyn Future<Output = ()> + 'static)>>`, but it returns `impl Future<Output = ()>`
         //~| ERROR expected `fn() -> impl Future<Output = ()> {callback}` to be a fn item that returns `Pin<Box<(dyn Future<Output = ()> + 'static)>>`, but it returns `impl Future<Output = ()>`
-        //~| ERROR expected `fn() -> impl Future<Output = ()> {callback}` to be a fn item that returns `Pin<Box<(dyn Future<Output = ()> + 'static)>>`, but it returns `impl Future<Output = ()>`
```

Comment:
It would be better if this PR was a pure refactoring not changing any logic, including diagnostics, if possible.
This PR is large and it's hard to notice and review any such changes.